src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateClone = Object.assign({}, state);
+  const history = [];
+
+  actions.forEach((action) => {
+    stateClone = Object.assign({}, stateClone);
+
+    switch (action.type) {
```

Comment:
Use destructuring for `action`
```suggestion
    stateClone = Object.assign({}, stateClone);
    
    const { type, ..... } = action;

    switch (type) {
```